schemas/components/starterHelper/StarterHelperView.tsx

```diff
@@ -0,0 +1,135 @@
+import {Image, Reference, SanityDocument} from '@sanity/types';
+import {Box, Button, Card, Flex, Stack, Text, TextArea, useToast} from '@sanity/ui';
+import React, {useCallback, useMemo} from 'react';
+// @ts-expect-error types not installed
+import styled from 'styled-components';
+// @ts-expect-error types not installed
+import {CopyToClipboard} from 'react-copy-to-clipboard';
+import imageUrlBuilder from '@sanity/image-url';
+import sanityClient from 'part:@sanity/base/client';
+import {CopyIcon} from '@sanity/icons';
+
+const client = sanityClient.withConfig({
+  apiVersion: '2023-01-01',
+});
+const urlBuilder = imageUrlBuilder(client);
+
+interface StarterTemplateDoc extends SanityDocument {
+  studioVersion?: number;
+  warningv2?: string;
+  title?: string;
+  description?: string;
+  slug?: {current?: string};
+  repository?: string;
+  demoURL?: string;
+  image?: Image;
+  authors?: Reference[];
+
+  //not used for studio v3
+  deploymentType?: string;
+  vercelDeployLink?: string;
+  repoId?: string;
+}
+
+export interface StarterHelperViewProps {
+  document?: {
+    displayed?: StarterTemplateDoc;
+  };
+}
+
+const RootBox = styled(Box)`
+  width: 100%;
+  height: 100%;
+  overflow: hidden;
+`;
+
+export function StarterHelperView({document}: StarterHelperViewProps) {
+  const doc = document?.displayed;
+
+  if (!doc || doc.studioVersion !== 3) {
+    return (
+      <RootBox padding={4}>
+        <Card padding={4}>Nothing of relevance to show.</Card>
+      </RootBox>
+    );
+  }
+
+  return (
+    <RootBox>
+      <Card padding={4}>
+        <VercelHelper doc={doc} />
+      </Card>
+    </RootBox>
+  );
+}
+
+export function VercelHelper({doc}: {doc: StarterTemplateDoc}) {
+  const toast = useToast();
+  const emitCopyToast = useCallback(() => {
+    toast.push({
+      status: 'info',
+      title: 'Vercel deploy link copied',
+    });
+  }, []);
+
+  const {title, description, repository, demoURL, image} = doc;
+
+  const vercelDeployLink = useMemo(() => {
+    let splitRepo = repository?.split('/');
+    const templateRepoName = splitRepo?.[splitRepo.length - 1];
+    const projectName = repository?.replace(/(template-|sanity-template-)/g, '');
+
+    const props = {
+      'repository-url': repository,
+      'project-name': projectName,
+      'repository-name': projectName,
+      'demo-title': title,
+      'demo-description': description,
+      'demo-url': demoURL,
+      'demo-image': image?.asset ? urlBuilder.image(image).url() : undefined,
+      'integration-ids': 'oac_hb2LITYajhRQ0i4QznmKH7gx',
+      'external-id': `nextjs;template=${templateRepoName}`,
```

Comment:
We recently implemented Vercel deploy functionality in one of our own SE studios, and to solve this part we have a optioned list of framework strings that Vercel supports. Perhaps we should put that in here as well?

Reply:
https://github.com/sanity-io/ix/blob/preview/schemas/documents/template.ts#L31-L72

Reply:
Sounds like a good idea!

Reply:
@runeb you should allow other: https://github.com/sanity-io/jumpstart/blob/01feea0de301784c18207d9f189120f45bb17943/src/frontend/screens/vercelIntegration/v2/frameworks.json#L208
This is necessary to support frameworks such as [Storybook](https://storybook.js.org/) 🙂

Reply:
List itself comes from https://www.npmjs.com/package/@vercel/frameworks btw

Reply:
After looking at the community studio data-structure, it seems to me like we should not use a static list for this, but should get the external-id from the framework reference value.

This will be a bit more involved than I care to fix in this PR. So instead we will remove the external-id from the link but have it just be ´;template=${templateRepoName}` which @stipsan assures me will work 😅